hw/ip/otbn/doc/_index.md

```diff
@@ -499,6 +523,107 @@ This signal is high when OTBN is not running.
 The cycle after a write to {{< regref "CMD.start" >}}, the signal goes low.
 This remains low until the end of the operation (either from an [`ECALL`]({{< relref "isa#ecall" >}}) or an error, at which point it goes high again.
 
+### Data Integrity Protection {#design-details-data-integrity-protection}
+
+OTBN stores and operates on data (state) in its dedicated memories, register files, and internal registers.
+OTBN's data integrity protection is designed to protected all data stored and transmitted within OTBN from modifications through physical attacks.
+
+During transmission, the integrity of data is protected with an integrity protection code.
+Data at rest in the instruction and data memories is additionally scrambled.
+
+In the following, the Integrity Protection Code and the scrambling algorithm are discussed, followed by their application to individual storage elements.
+
+#### Integrity Protection Code {#design-details-integrity-protection-code}
+
+OTBN uses the same integrity protection code everywhere to provide overarching data protection without regular re-encoding.
+The code is applied to 32b data words, and produces 39b of encoded data.
+
+The code used is an (39,32) Hsiao "single error correction, double error detection" (SECDED) error correction code (ECC) [[CHEN08]({{< relref "#ref-chen08">}})].
+It has a minimum Hamming distance of four, resulting in the ability to detect at least three errors in a 32 bit word.
+The code is used for error detection only; no error correction is performed.
+
+#### Memory Scrambling {#design-details-memory-scrambling}
+
+Contents of OTBN's instruction and data memories are scrambled while at rest.
+Both the data word itself and the address are scrambled.
```

Comment:
I can interpret this sentence two ways:

1) data is scrambled, and (separately) address is scrambled
2) data and address are together inputs to the scramble function (i.e. address affects the data written / ~binds the data to the specific address, and also affects the actual storage address in the memory)

I'm not sure which is the correct meaning here (I think it's 2?)

Reply:
It's 2. The keystream that gets XORed with the data is generated with an initial vector derived from the address and nonce. The address itself goes through the substitution/permutation network to form the physical memory address.

Reply:
I updated the text to give a bit more detail here without duplicating all the gory details, which are in the SRAM controller spec linked further down.